test/dns/query.cpp

```diff
@@ -99,6 +104,18 @@ TEST_CASE("throw error while fails evdns_set_options for attempts") {
     REQUIRE(false);
 }
 
+TEST_CASE("throw error while fails evdns_set_options for timeout") {
+    try {
+        create_evdns_base<::evdns_base_new, ::evdns_base_nameserver_ip_add,
+                          base_free_evdns_set_options_timeout>(
+            {{"dns/attempts", "nexa"}}, Reactor::global());
+    } catch (std::runtime_error &) {
+        REQUIRE(base_free_evdns_set_options_timeout_flag);
+        return;
+    }
```

Comment:
The problem was that I defined [this functions using the macro](https://github.com/measurement-kit/measurement-kit/pull/608/files/2a54277abae3fa3b3ba8ae678f92e0406e80f779#diff-8bb0df03508f949bf0f0057d91bdbf40R63), but it was never used, so I added a test to use it!